### DISCUSSION TITLE
Provide example for inserting a document using the docs1-cli

### DISCUSSION THREAD

**Discussion by jwmurray** [Byron/google-apis-rs#273] [Q&A] [answered]
Thank you for the very impressive group of API examples!

In using the docs1-cli example, I am able to create a new blank document.  However, I am not able to insert data into the document using the batch-update.  I have read through the google api docs, but do not understand how to access the features of the api.

Could you provide an example that would 

1. Insert text into a blank document
2. Replace text in a current document

Thank you!

**Comment by Byron** [comment, marked as answer]
Hi @jwmurray, it can happen that the way the API is made, it's hard or impossible to use certain features correctly. Here I am entirely unsure if that's the case.

As I have no time to provide this kind of support. My recommendation is to strip the CLI down to a minimal program that uses the API as you need it, which usually helps to track down potential shortcomings along with the required changes to make it work.

If you feel adventurous, you could also try [the new iteration of this repository](https://github.com/google-apis-rs/generated), as the API libraries should already be pretty functional - CLIs for these are not yet available though.

Thanks for your understanding

**Comment by jwmurray** [comment]
thank you for the pointers.  I will work on it.
